benchmark/kvbench/runtime/etcd_rt.py

```
@@ -168,70 +203,112 @@ def allgather_obj(self, obj: Any) -> List[Any]:
+            all_vals = []
+            start_time = time.time()
             for dest_rank in range(self.world_size):
                 val = self.client.get(f"{self.prefix}/all_reduce/{dest_rank}")[0]
-                vals.append(pickle.loads(val))
-
-            logger.debug("All reduce values: %s", vals)
+                while val is None:
+                    if time.time() - start_time > timeout_sec:
+                        raise TimeoutError(
+                            f"[Rank {self.rank}] all_reduce timeout waiting for rank {dest_rank} after {timeout_sec}s"
+                        )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)